lib/concepts/lesson/operations/create.rb

```diff
@@ -5,15 +5,25 @@ class Create
     class << self
       def call(lesson_params:)
         response = OperationResponse.new
-        response[:lesson] = Lesson.new(lesson_params)
+        response[:lesson] = build_lesson(lesson_params)
         response[:lesson].save!
         response
       rescue StandardError => e
         Sentry.capture_exception(e)
-        errors = response[:lesson].errors.full_messages.join(',')
-        response[:error] = "Error creating lesson: #{errors}"
+        response[:error] = "Error creating lesson: #{e}"
         response
       end
+
+      private
+
+      def build_lesson(lesson_hash)
+        new_lesson = Lesson.new(lesson_hash.except(:project_attributes))
+        project_params = lesson_hash[:project_attributes].merge({ user_id: lesson_hash[:user_id],
+                                                                  school_id: lesson_hash[:school_id],
+                                                                  lesson_id: new_lesson.id })
+        new_lesson.project = Project.new(project_params)
+        new_lesson
```

Comment:
🔧 I think it'd be best to do this as a [transaction](https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html) so we know both the lesson and project are created or its rolled back so neither are 👍